File: ungs-docsys-ms/src/main/java/com/ungs/docsys/controllers/UserInfoController.java
package com.ungs.docsys.controllers;

import com.ungs.docsys.dtos.UserInfoRequestDto;
import com.ungs.docsys.services.UserInfoService;
import io.swagger.v3.oas.annotations.Operation;
import io.swagger.v3.oas.annotations.responses.ApiResponse;
import jakarta.validation.Valid;
import lombok.AllArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/v1/users")
@AllArgsConstructor
public class UserInfoController {
    private final UserInfoService userInfoService;

    @Operation(summary = "Sign up a new user")
    @ApiResponse(responseCode = "201", description = "User created successfully")
    @ApiResponse(responseCode = "400", description = "Invalid input data")
    @ApiResponse(responseCode = "409", description = "User already exists")
    @PostMapping("/signUp")
    public ResponseEntity<Void> signUp(@RequestBody @Valid UserInfoRequestDto request) {
        userInfoService.signUp(request);
        return ResponseEntity.status(HttpStatus.CREATED).build();
    }
}

File: ungs-docsys-bff/src/modules/login/user-info/controllers/user-info.controller.ts
import { Controller, Post, Body } from "@nestjs/common";
import { UserInfoService } from "../services/user-info.service";
import { UserInfoRequestDto } from "../dtos/user-info-request.dto";
import { UserInfoResponseDto } from "../dtos/user-info-response.dto";
import { ApiOperation, ApiResponse } from "@nestjs/swagger";

@Controller('/v1/auth/register')
export class UserInfoController {
    constructor(private readonly userInfoService: UserInfoService) {}

    @Post()
    @ApiOperation({ summary: 'Sign Up' })
    @ApiResponse({
    status: 200,
        description: 'Successfully signed in',
        type: UserInfoResponseDto,
    })
    async signUp(@Body() request: UserInfoRequestDto): Promise<UserInfoResponseDto> {
        return await this.userInfoService.signUp(request);
    }
}